Tests in release version do not pass

The newly released v0.36.0 fails tests using Rust 1.85.0:

```console
$ cargo test --frozen --all-features
[...]
   Compiling comrak v0.36.0 (/build/comrak/src/comrak-0.36.0)
error[E0004]: non-exhaustive patterns: `&NodeValueDiscriminants::ShortCode` not covered
   --> src/tests/sourcepos.rs:423:30
    |
423 |             let text = match v {
    |                              ^ pattern `&NodeValueDiscriminants::ShortCode` not covered
    |
note: `NodeValueDiscriminants` defined here
   --> src/nodes.rs:16:25
    |
16  | #[cfg_attr(test, derive(strum::EnumDiscriminants))]
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^
...
166 |     ShortCode(NodeShortCode),
    |     --------- not covered
    = note: the matched value is of type `&NodeValueDiscriminants`
    = note: this error originates in the derive macro `strum::EnumDiscriminants` (in Nightly builds, run with -Z macro-backtrace for more info)
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
464 ~                 Raw => unreachable!(),
465 ~                 &NodeValueDiscriminants::ShortCode => todo!(),
    |
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -22,13 +22,13 @@ jobs:
     - name: Obtain Rust
       run: rustup override set ${{ matrix.rust }}
     - name: Build library
-      run: cargo build --verbose --lib
+      run: cargo build --locked --verbose --lib
     - name: Build examples
-      run: cargo build --verbose --lib --examples
+      run: cargo build --locked --verbose --lib --examples
     - name: Run unit tests
-      run: cargo test --verbose
-    - name: "Run README sample (TODO: update me)"
-      run: cargo run --example sample
+      run: cargo test --locked --verbose
+    - name: "Run README sample"
+      run: cargo run --locked --example sample
   build_bin_spec:
     runs-on: ubuntu-latest
     strategy:
@@ -45,7 +45,7 @@ jobs:
     - name: Obtain Rust
       run: rustup override set ${{ matrix.rust }}
     - name: Build binary
-      run: cargo build --verbose --bin comrak --release
+      run: cargo build --locked --verbose --bin comrak --release
     - name: Run spec tests
       run: script/cibuild
   build_wasm:
@@ -66,9 +66,9 @@ jobs:
     - name: Setup for wasm
       run: rustup target add wasm32-unknown-unknown
     - name: Build
-      run: cargo build --verbose --target wasm32-unknown-unknown
+      run: cargo build --locked --verbose --target wasm32-unknown-unknown
     - name: Build examples
-      run: cargo build --verbose --target wasm32-unknown-unknown --examples
+      run: cargo build --locked --verbose --target wasm32-unknown-unknown --examples
   no_features_build_test:
     runs-on: ubuntu-latest
     strategy:
@@ -85,17 +85,24 @@ jobs:
     - name: Obtain Rust
       run: rustup override set ${{ matrix.rust }}
     - name: Build and test with no features
-      run: cargo test --no-default-features --tests
-  lockfile:
+      run: cargo test --locked --no-default-features
+  all_features_build_test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - nightly
+          - beta
+          - stable
+          - $MSRV
     steps:
     - uses: actions/checkout@v4
       with:
         submodules: true
     - name: Obtain Rust
-      run: rustup override set $MSRV
+      run: rustup override set ${{ matrix.rust }}
     - name: Build and test with no features
-      run: cargo +stable build --locked --release --all-features
+      run: cargo test --locked --all-features
   clippy_format:
     runs-on: ubuntu-latest
     steps:

--- a/src/tests/sourcepos.rs
+++ b/src/tests/sourcepos.rs
@@ -185,6 +185,14 @@ const FOOTNOTE_REFERENCE: TestCase = (
 "#,
 );
 
+#[cfg(feature = "shortcodes")]
+const SHORTCODE: TestCase = (
+    &[sourcepos!((2:2-2:7))],
+    r#"nya~
+!:fire:!
+"#,
+);
+
 const TABLE: TestCase = (
     &[sourcepos!((3:1-5:17))],
     r#"stuff before
@@ -440,6 +448,8 @@ fn node_values() -> HashMap<NodeValueDiscriminants, TestCase> {
                 ThematicBreak => THEMATIC_BREAK,
                 FootnoteDefinition => FOOTNOTE_DEFINITION,
                 FootnoteReference => FOOTNOTE_REFERENCE,
+                #[cfg(feature = "shortcodes")]
+                ShortCode => SHORTCODE,
                 Table => TABLE,
                 TableRow => TABLE_ROW,
                 TableCell => TABLE_CELL,
@@ -485,6 +495,10 @@ fn sourcepos() {
     options.extension.superscript = true;
     options.extension.subscript = true;
     options.extension.autolink = true;
+    #[cfg(feature = "shortcodes")]
+    {
+        options.extension.shortcodes = true;
+    }
     options.extension.math_code = true;
     options.extension.math_dollars = true;
     options.extension.multiline_block_quotes = true;

